src/assocPath.js

```diff
@@ -29,6 +29,8 @@ var isNil = require('./isNil');
  *
  *      // Any missing or non-object keys in path will be overridden
  *      R.assocPath(['a', 'b', 'c'], 42, {a: 5}); //=> {a: {b: {c: 42}}}
+ *
+ *      R.assocPath([-1], 42, [1,2,3]); //=> [1,2,42]
  */
 module.exports = _curry3(function assocPath(path, val, obj) {
   if (path.length === 0) {
```

Comment:
github won't let me comment on a line that isn't part of the diff, but I think we need some logic here to handle multiple negative indexes (eg `assocPath([-1, -1], 10, [[1, 2, 3], [4, 5, 6]])`):

https://github.com/keithbro/ramda/blob/d8bb8c71e70ca26d74aec1ffeafd4e6eabc80b17/src/assocPath.js#L40-L43

let's add a test for this as well